src/index.ts

```diff
@@ -261,18 +264,27 @@ class Envoy {
 
 
           app_state.desired_activation_state = 'deactivated'
+          log.normal('setting desired state to deactivated. actual: %s', this.app_states[installed_app_id].desired_activation_state)
           app_state.desired_activation_state_changed_at = Date.now()
 
 
+          const when_this_interval = Date.now().toLocaleString()
           let deactivationInterval = setInterval(() => {
+            log.normal("trying to deactivate. desired_activation_state: %s", app_state.desired_activation_state)
+            log.normal("this interval initiated at: %s", when_this_interval)
+            log.normal("app_state: %s", inspect(app_state))
+
             if (app_state.desired_activation_state === 'activated') {
+              log.normal("clearing deactivation interval initiated at: %s because of something in app_state", when_this_interval)
+
               clearInterval(deactivationInterval)
               return
             }
 
+            // dont deactivate too soon or if currently transitioning
             if (app_state.activation_state_changed_at + 5000 >= Date.now() ||
-                app_state.activation_state !== 'activated' ||
-                app_state.desired_activation_state_changed_at + 5000 >= Date.now()
+                app_state.desired_activation_state_changed_at + 5000 >= Date.now() ||
+                app_state.desired_activation_state.endsWith('ing')
```

Comment:
Curious about why we don't just explicitly state `app_state.activation_state === 'activating'` here instead as it looks like we're only ever setting `app_state.activation_state = 'deactivating'` after this condition check, leaving the `'activating'` state as the only possible state ending with 'ing' at the time of this check. (PS. I'm also assuming  `activation_state` was meant to be written here instead of `desired_activation_state` , as `desired_activation_state` doesn't include the present participle actions as its value.) 
```suggestion
                app_state.activation_state === 'activating'
```

Reply:
I think you're missing that these are in an interval. The activation_state can definitely be `'deactivating'` at this point. Good point about the variable name though

Reply:
True they're in a loop, ...and it's definitely safer to assume the call could take longer than 5 seconds. :+1: Nevermind, the first point then.